Bug-Debian: http://bugs.debian.org/582163

If we get an error SNAC on the ICBM family and it's missing buddy name then
don't fallthrough to the default error handler in misc.c. This was causing
purple_parse_msgerr() in oscar.c to get called with different va_args than
it was expecting, which caused a crash. Specifically when trying to fetch
the ICQ x-status of an offline buddy.
#
#
# patch "libpurple/protocols/oscar/family_icbm.c"
#  from [fcbebddf7bb56d37e2e89ffa10da1b73e63cf833]
#    to [62126e23e6971a85d9b599c7ca61675a008e6f9b]
#
============================================================
--- pidgin.orig/libpurple/protocols/oscar/family_icbm.c	fcbebddf7bb56d37e2e89ffa10da1b73e63cf833
+++ pidgin.org/libpurple/protocols/oscar/family_icbm.c	62126e23e6971a85d9b599c7ca61675a008e6f9b
@@ -166,20 +166,20 @@ error(OscarData *od, FlapConnection *con
 
 	if (!(snac2 = aim_remsnac(od, snac->id))) {
 		purple_debug_misc("oscar", "icbm error: received response from unknown request!\n");
-		return 0;
+		return 1;
 	}
 
 	if (snac2->family != SNAC_FAMILY_ICBM) {
 		purple_debug_misc("oscar", "icbm error: received response from invalid request! %d\n", snac2->family);
 		g_free(snac2->data);
 		g_free(snac2);
-		return 0;
+		return 1;
 	}
 
 	if (!(bn = snac2->data)) {
 		purple_debug_misc("oscar", "icbm error: received response from request without a buddy name!\n");
 		g_free(snac2);
-		return 0;
+		return 1;
 	}
 
 	reason = byte_stream_get16(bs);
